apps/govquests-frontend/src/domains/action_tracking/components/ActionHandler.tsx

```diff
@@ -0,0 +1,59 @@
+// src/domains/actions/components/ActionHandler.tsx
+import React, { useState } from "react";
+import { ActionExecution } from "../types/actionTypes";
+import { Action } from "@/domains/questing/types/questTypes";
+import { useStartActionExecution } from "../hooks/useStartActionExecution";
+import { useCompleteActionExecution } from "../hooks/useCompleteActionExecution";
+import { ActionStrategyFactory } from "../strategies/ActionStrategyFactory";
+import { useSignMessage } from "wagmi";
+
+interface ActionHandlerProps {
+  action: Action;
+}
+
+const ActionHandler: React.FC<ActionHandlerProps> = ({ action }) => {
+  const [execution, setExecution] = useState<ActionExecution | null>(null);
+  const startMutation = useStartActionExecution();
+  const completeMutation = useCompleteActionExecution();
+  const strategy = ActionStrategyFactory.createStrategy(action.actionType);
+  const { signMessage } = useSignMessage();
+
+  const handleStart = async () => {
+    await strategy.start(startMutation, action);
+    setExecution(
+      startMutation.data?.startActionExecution?.actionExecution || null
+    );
+  };
+
+  const handleComplete = async () => {
+    if (!execution) return;
+    let completionData = {};
+    if (action.actionType === "gitcoin_score") {
+      const signature = await signMessage({
+        message: execution.startData.message,
+      });
+      completionData = { address: execution.startData.address, signature };
+    }
+    await strategy.complete(completeMutation, execution, completionData);
+  };
+
+  if (startMutation.isLoading || completeMutation.isLoading) {
```

Comment:
I think that in latest versions of React Query it should  be `isPeding` instead of `isLoading`.